homeassistant/components/knx/weather.py

```diff
@@ -23,20 +36,21 @@ class KNXWeather(KnxEntity, WeatherEntity):
 
     def __init__(self, device: XknxWeather):
         """Initialize of a KNX sensor."""
+        self._device: XknxWeather
         super().__init__(device)
 
     @property
-    def temperature(self):
+    def temperature(self) -> float | None:
         """Return current temperature."""
         return self._device.temperature
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return temperature unit."""
         return TEMP_CELSIUS
 
     @property
-    def pressure(self):
+    def pressure(self) -> float | None:
```

Comment:
Side note; not for this PR: this can be guarded instead of an inline if/else construct.

```py
if not self._device.air_pressure:
  return None
return self._device.air_pressure / 100
```